clisops/core/regrid.py

```diff
@@ -0,0 +1,1516 @@
+"""Regrid module."""
+import warnings
+from pathlib import Path
+from typing import Tuple, Union
+
+import cf_xarray as cfxr
+import numpy as np
+import roocs_grids
+import scipy
+import xarray as xr
+
+try:
+    import xesmf as xe
+except ImportError:
+    raise ValueError(
+        "Package xesmf >= 0.6.0 is required to use the regridding functionality."
+    )
```

Comment:
The xesmf import here causes problems now with the CI. How to deal with this?
In the core/__init__.py one can try to import xesmf, and if not possible, not import anything from core/regrid.py?

Reply:
@sol1105 I have made a change to how we manage the import of `xesmf`. It uses a decorator that we attach to the main functions and `__init__()` constructors for classes that require `xesmf`, e.g.:

```
@require_xesmf
def regrid(ds, regridder, adaptive_masking_threshold=0.5):
```

I am still trying to understand where the other failures are coming from in the CI.